avocado/utils/software_manager/manager.py

```diff
@@ -49,3 +49,24 @@ def is_capable(self):
         except NotImplementedError:
             pass
         return self.initialized
+
+    @staticmethod
+    def extract_from_package(package_path, dest_path=None):
+        """Try to extract a package content into a destination directory.
+
+        It will try to see if the package is valid against all supported
+        package managers and if any is found, then extracts its content into
+        the extract_path.
+
+        Raises NotImplementedError when a non-supported package is used.
+
+        :param str package_path: package file path.
+        :param str dest_path: destination path to extract. Default is the
+                              current directory.
+        :returns: destination path were the package it was extracted.
+        """
+        for backend in SUPPORTED_PACKAGE_MANAGERS.values():
```

Comment:
Sorry for the late change request. You don't need to do it here. You need to `_init_on_demand` to have the `self.backend` set correctly and then just use it.

Reply:
As mentioned on the commit message I'm not using it because it is distro based. It will set the backend depending on the distro. But here we my intention it was to be able to extract packages from cross distros, if we have the tooling.

Reply:
Okay, that makes sense, thanks!